src/Core/Content/Media/DataAbstractionLayer/MediaIndexer.php

```diff
@@ -119,17 +187,43 @@ public function handle(EntityIndexingMessage $message): void
                 'id' => Uuid::fromHexToBytes($id),
             ]);
         }
-
-        $this->eventDispatcher->dispatch(new MediaIndexerEvent($ids, $context, $message->getSkip()));
     }
 
-    public function getTotal(): int
+    private function setMediaPaths(Context $context, array $ids): void
     {
-        return $this->iteratorFactory->createIterator($this->repository->getDefinition())->fetchCount();
-    }
+        $mediaIdsWithMissingPaths = $this->connection->fetchFirstColumn(
+            'SELECT LOWER(HEX(id)) from media WHERE id IN (:ids) AND path IS NULL AND file_name IS NOT NULL',
+            ['ids' => Uuid::fromHexToBytesList($ids)],
+            ['ids' => Connection::PARAM_STR_ARRAY]
+        );
 
-    public function getDecorated(): EntityIndexer
-    {
-        throw new DecorationPatternException(static::class);
+        if (\count($mediaIdsWithMissingPaths) === 0) {
+            return;
+        }
+
+        $query = new RetryableQuery(
+            $this->connection,
+            $this->connection->prepare('UPDATE `media` SET path = :path WHERE id = :id')
+        );
+
+        //get all media with missingPaths
+        $criteria = new Criteria();
```

Comment:
no dal pure sql

Reply:
no, I need the mediaEntity for the pathGenerator :-)